src/main/java/com/microsoftopentechnologies/windowsazurestorage/AzureBlobAction.java

```diff
@@ -1,42 +1,39 @@
 package com.microsoftopentechnologies.windowsazurestorage;
 
+import com.microsoftopentechnologies.windowsazurestorage.WAStoragePublisher.WAStorageDescriptor;
+import com.microsoftopentechnologies.windowsazurestorage.beans.StorageAccountInfo;
+import hudson.model.Run;
+import hudson.model.RunAction;
 import java.io.IOException;
 import java.util.List;
-
 import javax.servlet.ServletException;
-
 import jenkins.model.Jenkins;
-
 import org.acegisecurity.Authentication;
 import org.kohsuke.stapler.StaplerRequest;
 import org.kohsuke.stapler.StaplerResponse;
 
-import hudson.model.AbstractBuild;
-import hudson.model.RunAction;
-import hudson.model.Run;
-
-import com.microsoftopentechnologies.windowsazurestorage.WAStoragePublisher;
-import com.microsoftopentechnologies.windowsazurestorage.WAStoragePublisher.WAStorageDescriptor;
-import com.microsoftopentechnologies.windowsazurestorage.beans.StorageAccountInfo;
-
 
 public class AzureBlobAction implements RunAction {
-	private final AbstractBuild build;
 	private final String storageAccountName;
 	private final String containerName;
 	private final boolean allowAnonymousAccess;
 	private final AzureBlob zipArchiveBlob;
 	private final List<AzureBlob> individualBlobs;
+	private final Run build;
 
-	public AzureBlobAction(AbstractBuild build, String storageAccountName, String containerName,
```

Comment:
This is typically how this is done without storing a pointer to the BUild:

https://github.com/jenkinsci/build-failure-analyzer-plugin/blob/master/src/main/java/com/sonyericsson/jenkins/plugins/bfa/model/FailureCauseProjectAction.java#L41

You have two actions, one for projects and one for runs.  Runs are just attached normally and contain the data.  Project actions dynamically retrieve the last build and then the action from that build if it exists.  This avoids any back points which cause havoc with the GC.

/cc @crummel